transitions/index.js

```diff
@@ -214,40 +220,33 @@ var applyTransition = function(options, callback) {
 
     /*
      * Transitions are async and should return true if the document has
-     * changed.  If a transition errors then log the error, but don't
-     * return it because that will stop the series and the other
-     * transitions won't run.
+     * changed.  If a transition errors then log the error, but don't return it
+     * because that will stop the series and the other transitions won't run.
      */
     transition.onMatch(change, db, audit, function(err, changed) {
         logger.debug(
-            'finished transition for doc %s seq %s transition %s',
-            change.id, change.seq, key
+            'finished transition %s for seq %s doc %s is %s',
+            key, change.seq, change.id, changed ? 'changed' : 'unchanged'
         );
-        // todo?
-        // prop = doc.transitions && doc.transitions[key] ? doc.transitions[key] : {};
-        //_setProperty('tries', prop.tries ? prop.tries++ : 1)
-        //_setProperty('couch', 'uuid');
-        //
         if (err || changed) {
```

Comment:
It hurts in this case because if a transition that returns an error is followed by a different transition that doesn't have an error then the first transition will never run again. This is exactly the case this PR is trying to solve. We could change all `hasRun` functions to check that `ok=true`, or not store the error. I'll have a look.